lib/topological_inventory/ansible_tower/operations/worker.rb

```diff
@@ -71,6 +73,16 @@ def default_messaging_opts
             :group_ref  => "topological-inventory-operations-ansible-tower"
           }
         end
+
+        # TODO: Probably move this to common eventually, if we need it elsewhere.
+        def heartbeat_thread(client = nil)
+          @heartbeat_thread ||= Thread.new do
+            loop do
+              sleep(15)
+              client.send(:topic_consumer, queue_opts[:persist_ref], nil).trigger_heartbeat!
```

Comment:
it should catch a recover from potential exception

Reply:
Updated!